torch/_subclasses/fake_impls.py

```diff
@@ -269,6 +268,11 @@ def unique2(
         # Without symints/symfloats, cannot handle this
         raise DynamicOutputShapeException(func)
 
+    if dim is None:
+        arg_dim = arg
+    else:
+        arg_dim = arg.new_empty((arg.shape[dim],))
+
     if (nnz := arg.unique_memo) is None:
```

Comment:
The memo strategy needs to be adjusted here, because the unique at some dimension differs from uniques at other dims. You are sharing the memo for everything.

Reply:
I'll admit I haven't dug deep enough to understand what is going on with the memos yet.

Are the memos even necessary? [This comment](https://github.com/pytorch/pytorch/pull/121571#discussion_r1529510475) implies to me that they may not be needed for unique.

Reply:
I think the simplest resolution here is to just not use the memo for unique dim

Reply:
Adopted the suggested resolution in e040b23